pandas/plotting/_core.py

```diff
@@ -2938,26 +2938,59 @@ def scatter(self, x, y, s=None, c=None, **kwds):
     def hexbin(self, x, y, C=None, reduce_C_function=None, gridsize=None,
                **kwds):
         """
-        Hexbin plot
+        Make hexagonal binning plot.
+
+        Make an hexagonal binning plot of `x` versus `y`. If `C` is `None`
+        (the default), this is an histogram of the number of occurrences
+        of the observations at (x[i],y[i]).
+
+        If `C` is specified, specifies values at given coordinates
+        (x[i],y[i]). These values are accumulated for each hexagonal
+        bin and then reduced according to `reduce_C_function`,
+        having as default the numpy's mean function (np.mean).
+        (If `C` is specified, it must also be a 1-D sequence
+        of the same length as `x` and `y`.)
 
         Parameters
         ----------
-        x, y : label or position, optional
-            Coordinates for each point.
+        x : label or position
```

Comment:
In the guide we ask people to use types like, `int`, `str` here for the parameter types. Should we accept "label or position" as well for types for column selection? This is something very common in many docstrings and we should to an agreement.

cc/ @datapythonista , @jorisvandenbossche

Reply:
In my opinion I'd use `int or str`, and use the description to say it's the label or the position. I don't see a reason to make an exception here.I think it's useful for the user to know that it's `int` or `str`. And who knows if in the future having the types can be useful in a way similar to mypy. :)

Reply:
The reason we *might* make an exception is because 'str' is too restrictive for labels, as a column labels can in principle be any hashable object ... (but ok, in most cases str will be what people use)

But I am fine here with using `int or str`

Reply:
Yes, I forgot about the type. I have a doubt with `reduce_C_function` parameter: it only accepts numpy functions in the form `np.mean` or  `np.std`  not strings like `'mean'`. Is  `function` (or `func`) the correct type reference?